File: algorithm/sort/select_sort/select_sort.c
#include <stdio.h>
#include <stdlib.h>

void array_print(int *nums, int len) {
    for (int i = 0; i < len; i++) {
        printf("%d ", nums[i]);
    }
    printf("\n");
}

void swap(int *a, int *b) {
    int tmp = *a;
    *a = *b;
    *b = tmp;
}

int main(int argc, char *argv[]) {
    if (argc < 2) {
        return 0;
    }
    int len = atoi(argv[1]);
    int *nums = (int *) malloc(len * sizeof(int));
    for (int i = 0; i < len; i++) {
        nums[i] = rand() % 300;
    }
    array_print(nums, len);

    for (int i = 0; i < len; i++) {
        int min_val = nums[i];
        int min_index = i;
        for (int j = i; j < len; j++) {
            if (nums[j] < min_val) {
                min_val = nums[j];
                min_index = j;
            }
        }
        if (min_index != i) {
            swap(&nums[i], &nums[min_index]);
        }
    }
    array_print(nums, len);
    free(nums);
    return 0;
}

File: algorithm/sort/select_sort/SelectSort.java
import java.util.Arrays;
import java.util.Random;

public class SelectSort {
    public static void main(String[] args) {
        if (args.length < 1) {
            return;
        }
        int len = Integer.parseInt(args[0]);
        int[] nums = new int[len];
        Random random = new Random();
        for (int i = 0; i < nums.length; i++) {
            nums[i] = random.nextInt(300);
        }
        System.out.println(Arrays.toString(nums));
        sort(nums);
        System.out.println(Arrays.toString(nums));
    }

    public static void sort(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            int min_val = nums[i];
            int min_index = i;
            for (int j = i; j < nums.length; j++) {
                if (nums[j] < min_val) {
                    min_val = nums[j];
                    min_index = j;
                }
            }
            if (min_index != i) {
                swap(nums, min_index, i);
            }
        }
    }

    private static void swap(int[] nums, int i, int j) {
        int tmp = nums[i];
        nums[i] = nums[j];
        nums[j] = tmp;
    }
}
